src/LtiLineitem.php

```diff
@@ -143,4 +146,16 @@ public function setEndDateTime($value)
 
         return $this;
     }
+
+    public function getGradesReleased()
+    {
+        return $this->grades_released;
+    }
+
+    public function setGradesReleased($value)
```

Comment:
Is it possible to add typing to this variable? I looked at the spec, and it doesn't offer any indication as to what type(s) are allowed here. Is this a boolean, string, timestamp, black hole, something else?

Edit: I found some examples in the spec doc, and it looks like they're using bools.

```suggestion
    public function setGradesReleased(bool $value)
```

Reply:
Its not called out in the spec, but the examples on the ags spec show it as a boolean. It would have to be parsed as such by us here. @JonathanGawrych is with me at GoReact and we noticed the `gradesReleased` property was added this january to the spec, so we'd like to just get ahead of that and build in support now.

Reply:
I was following the pattern of the other properties, not adding type information. Indeed the tests even pass the string 'expected' to every property, even numbers like getScoreMaximium. I added it without types because I wasn't sure which version of php was supported. It's not in the composer.json

I think it would be good if we did a seperate MR to add typing information, what do you think?. I see the tests are running 7.4. Something like activityProgress would be good as a string enum, as the only values are Initialized, Started, InProgress, Submitted, or Completed. Are there any plans to move to PHP >=8.1?

Reply:
Yeah, I've got some work to do first, but my goal is to set the minimum to >=8.1 and strongly type the package (to the extent that the LTI spec allows, at least).

Reply:
Sorry for the delay. I've added typing information. Thanks!